test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -788,6 +1006,8 @@ def stopTest(self, test):
         if old_sigalrm_handler is not None:
             signal.signal(signal.SIGALRM, old_sigalrm_handler)
             signal.alarm(0)
+        if self.timer_start:
```

Comment:
I suggest moving this if condition in `self.hardware_report()`, then you can get rid of `self.timer_start`

Reply:
I have to stop timer at the time of test end. In above I am checking, timer is started then true and timer cancel at test end otherwise no need to take any action.
According to you i have to monitor test end and stop timer. That is reverse of above.

Reply:
To stop the timer at test end you can call self.hardware_report_timer.cancle().

Reply:
In stopTest i am stopping timer if self.timer_start(timer is started). If i don't use timer monitoring in stopTest function then timer continue to run even the test end successfully.
if i use if condition in hardware_report function then i have to monitor when test is going to end. so it is reverse of the above.
anyhow i have to use one variable which can monitor timer_start/test_end also one if block.

Reply:
I understand the use case so you call self.hardware_report_timer.cancel() then the timer will be stopped forcefully so I don't think we need self.timer_start.